browser/brave_ads/tooltips/ads_captcha_tooltip.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/tooltips/ads_captcha_tooltip.h"
+
+#include "brave/components/brave_adaptive_captcha/buildflags/buildflags.h"
+
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+#include "brave/components/brave_adaptive_captcha/brave_adaptive_captcha_service.h"
+#endif
+
+namespace brave_ads {
+
+const char kScheduledCaptchaTooltipId[] = "scheduled-captcha";
+
+AdsCaptchaTooltip::AdsCaptchaTooltip(
+    brave_adaptive_captcha::BraveAdaptiveCaptchaService*
```

Comment:
I think it would be better to pass a WeakPtr callback for `ShowScheduledCaptcha`

Reply:
Good point, added that.